Content.Shared/_RMC14/Atmos/SharedRMCFlammableSystem.cs

```diff
@@ -339,6 +342,171 @@ private void OnFlammableIgnite(Entity<FlammableComponent> ent, ref RMCIgniteEven
     private void OnFlammableExtinguished(Entity<FlammableComponent> ent, ref RMCExtinguishedEvent args)
     {
         RemCompDeferred<OnFireComponent>(ent);
+        if (IsEntityTouchingFire(ent.Owner))
+        {
+            _entitiesNeedingRecheck.Add(ent.Owner);
+        }
+    }
+
+    private void OnIgniteStartCollide(Entity<RMCIgniteOnCollideComponent> ent, ref StartCollideEvent args)
+    {
+        if (!ShouldProcessIgniteCollision(ent.Comp, args.OurFixture, args.OtherFixture))
+            return;
+
+        TrackFireContact(ent.Owner, args.OtherEntity);
+        TryIgnite(ent, args.OtherEntity, false);
+
+        if (_flammableQuery.HasComp(args.OtherEntity))
+        {
+            EnsureComp<SteppingOnFireComponent>(args.OtherEntity);
+            _entitiesNeedingRecheck.Add(args.OtherEntity);
+        }
+    }
+
+    private void OnIgniteEndCollide(Entity<RMCIgniteOnCollideComponent> ent, ref EndCollideEvent args)
+    {
+        UntrackFireContact(ent.Owner, args.OtherEntity);
+
+        if (!IsEntityTouchingFire(args.OtherEntity))
+        {
+            RemCompDeferred<SteppingOnFireComponent>(args.OtherEntity);
+            _entitiesNeedingRecheck.Remove(args.OtherEntity);
+        }
+    }
+
+    private void OnIgniteDamageCollide(Entity<RMCIgniteOnCollideComponent> ent, ref DamageCollideEvent args)
+    {
+        Ignite(args.Target, ent.Comp.Intensity, ent.Comp.Duration, ent.Comp.MaxStacks);
+    }
+
+    private void OnIgniteComponentInit(Entity<RMCIgniteOnCollideComponent> ent, ref ComponentInit args)
+    {
+        Timer.Spawn(TimeSpan.FromMilliseconds(50), () => CheckForExistingOverlaps(ent));
```

Comment:
Is the timer here necessary? They are typically bad form, since they don't serialize properly and depending on test execution might make them break as well
If this is to wait until component data is valid, you might want to try using MapInitEvent instead